app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -161,4 +225,32 @@ private void addItemsOnSocialSitualtionSpinner(){
         socialSituationSpinner.setAdapter(socialSituationAdapter);
     }
 
+    // get the selected dropdown list value
+    public void addListenerOnSubmitButton() {
+
+        emotionalStateSpinner = (Spinner) findViewById(R.id.emotional_state_spinner);
+        socialSituationSpinner = (Spinner) findViewById(R.id.social_event_spinner);
+        Button submitButton = (Button) findViewById(R.id.post_mood_button);
+
+        submitButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+
+                emotionalState = String.valueOf(emotionalStateSpinner.getSelectedItem());
+                socialSit = String.valueOf(socialSituationSpinner.getSelectedItem());
+
+                try {
+                    submitMood();
+                    Toast.makeText(getApplicationContext(), "Successfully posted!", Toast.LENGTH_SHORT).show();
+                    finish();
+                } catch (DescriptionTooLongException e) {
+                    Toast.makeText(getApplicationContext(), "Your description is too long.", Toast.LENGTH_SHORT).show();
```

Comment:
If the user puts in more than 3 words, we throw an exception. Three words isn't much for a mood description. We throw this same exception if the user puts in more than 20 characters, spaces included. This sentence has more than 20 chars in it. It also has more than 3 words. Why limit the user to such a small description?